File: frontend/types/job.ts
export type JobStatus = "pending" | "processing" | "completed" | "failed";

export interface JobInput {
  file_id: string;
  filename: string;
  detected_type: string;
  size_bytes: number;
}

export interface JobOutput {
  filename?: string;
  size_bytes?: number;
  mime_type?: string;
  download_url?: string;
  preview?: JobPreview;
}

export interface JobPreview {
  /** preview type, e.g. "text", "image", "image_gallery", "pdf_thumbnail", "generic" */
  type: string;
  /** arbitrary data for the specific preview component */
  data: any;
}

export interface Job {
  job_id: string;
  tool_id: string;
  status: "pending" | "processing" | "completed" | "failed";
  created_at?: string;
  updated_at?: string;
  inputs?: any[]; // keep generic, not used directly in UI
  options?: Record<string, unknown>;
  output?: JobOutput;
  error?: any; // can be string or object from backend
}

File: backend/app/models/job.py
from datetime import datetime
from typing import Any, Dict, List, Optional
from pydantic import BaseModel, Field


class JobInput(BaseModel):
    file_id: str
    filename: str
    detected_type: str
    size_bytes: int
    temp_path: str | None = None


class JobOutput(BaseModel):
    filename: Optional[str] = None
    download_url: Optional[str] = None
    size_bytes: Optional[int] = None


class JobStatus:
    PENDING = "pending"
    PROCESSING = "processing"
    COMPLETED = "completed"
    FAILED = "failed"


class Job(BaseModel):
    job_id: str
    tool_id: str
    status: str = JobStatus.PENDING
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)
    inputs: List[JobInput] = []
    options: Dict[str, Any] = {}
    output: Optional[JobOutput] = None
    error: Optional[str] = None